src/ray/rpc/gcs_server/gcs_rpc_client.h

```diff
@@ -225,14 +229,42 @@ class GcsRpcClient {
   }
 
   void Reconnect() {
+    absl::MutexLock lock(&mutex_);
     if (get_server_address_) {
-      auto address = get_server_address_();
-      Init(address.first, address.second, client_call_manager_);
+      std::pair<std::string, int> address;
+      int index = 0;
+      for (; index < RayConfig::instance().ping_gcs_rpc_server_max_retries(); ++index) {
+        address = get_server_address_();
```

Comment:
print a debug log here: "Attempt to reconnect to GCS server: xxx".

Reply:
fixed